lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module

```diff
@@ -60,21 +60,39 @@ function dosomething_rogue_send_reportback_to_rogue($values, $user = NULL) {
 
   $client = dosomething_rogue_client();
 
+  // If there is no file, check to see if the user has already reported back for this campaign in Phoenix.
+  if (! isset($values['file'])) {
+    $rbid = dosomething_reportback_exists($values['nid'], $run->nid, $user->uid);
+    // If there is a RB in Phoenix, check the dosomething_rogue_reportback table to see if the RB already exists in Rogue.
+    if ($rbid) {
+      $exists_in_rogue = db_query("SELECT * FROM {dosomething_rogue_reportbacks} rogue_rbs WHERE rbid = :rbid", array(':rbid' => $rbid))->fetchAll();
+      // If the RB does not yet exist in Rogue, send along the most recently created reportback_item so the RB can be created in Rogue.
+      if (! $exists_in_rogue) {
+        $reportback = reportback_load($rbid);
+        $most_recent_rb_item_id = array_pop($reportback->fids);
+        $reportback_file = entity_load('reportback_item', array($most_recent_rb_item_id));
+        $image = file_load($most_recent_rb_item_id);
+        $values['file'] = 'data: ' . mime_content_type($image->filemime) . ';base64,' . base64_encode(file_get_contents($image->uri));
```

Comment:
ah ok well then this is why I made that other comment that we should be doing this logic in the submit hook. It seems inefficient to duplicate logic.